### ISSUE TITLE
Select Call Center - September Billing Files

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#489]
In GitLab by @kdjstudios on Sep 25, 2017, 12:58

Hello Team,

To help keep things organized I have started a new ticket to maintain progress on the September Billing files. Keep in mind this now puts us 6 months without converting the client.

* [SEP20_17_b2cmr1.csv](/uploads/98567445e6839415abfac67e6b6dfa6e/SEP20_17_b2cmr1.csv)
* [Invoicing_details_.xlsx](/uploads/9f34a4da1064594a3a992285118bf94f/Invoicing_details_.xlsx) - **DO NOT USE**
* [SEP20_17_b2tas1.csv](/uploads/c2cd137e789977f350a124730ba9c384/SEP20_17_b2tas1.csv)
* [SEP20_17_b2tas2.csv](/uploads/78c58f2a29e5191ccc030de7a5041dd1/SEP20_17_b2tas2.csv)
* [SEP20_17_b2rcr1.csv](/uploads/7b41e59e30af0ec7bdb17cde90e907aa/SEP20_17_b2rcr1.csv)

Updated files:

* [Sept20-invoicing-details._.pdf](/uploads/a71dddaa1bb031fb8b3fe0e2a5db4293/Sept20-invoicing-details._.pdf)
* [Sept20-invoicing-details._.xlsx](/uploads/5738560e5b066c47ff2a4de2287e99a2/Sept20-invoicing-details._.xlsx) - Converted using https://www.pdftoexcel.com/

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 25, 2017, 13:01

@smasih @tim.traylor @abakshi @pchaudhary 

This is in continuation of the ticket #468 and #305 ; This is the next billing cycle that the client has had to complete in their billing system. Once we have resolved issues with the current import and processed the last months cycle in comparison we can start to discuss to run this cycle after or start over fresh and reimport and run a comparison.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 7, 2017, 24:10

@kdjstudios 

Sorry but the usage file is missing. Please send it over.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 7, 2017, 13:19

@smasih 

I have reached out to Garrett for their September usage file and will post it here once received.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 7, 2017, 23:24

thanks :smiley:

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 9, 2017, 10:04

Here is the usage file:

[SEP20_17_TAE.csv](/uploads/213409956847f02fddd48d175423c788/SEP20_17_TAE.csv)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 9, 2017, 10:14

It appears this is in a different format from the previous usage files. I responded to Garrett with the follow:

Thank you for the usage file. However, it appears this is in a different format from the previous usage files you have sent us. Is there any chance you could continue to send us the file in the same format as before? As this is how the support team has programmed the import of the usage around that file formatting. I have attached a previous usage file that was sent to us.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 9, 2017, 11:15

@smasih @tim.traylor @kace.coady 

[TUSACC10.csv](/uploads/716c51ebf2adf0f55704b438dd3f103d/TUSACC10.csv)

He sent over this version. However when I went into Dev and tried to process this file in a new Billing Cycle I recieved the "Something went wrong error"

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 10, 2017, 04:45

@tim.traylor

Can we use this version [SEP20_17_TAE.csv](http://gitlab.aavaz.biz/AnswerNet/SABilling/uploads/213409956847f02fddd48d175423c788/SEP20_17_TAE.csv) as this one has already combined 5 rows into one and would save us lot of time compared to [TUSACC10.csv](http://gitlab.aavaz.biz/AnswerNet/SABilling/uploads/716c51ebf2adf0f55704b438dd3f103d/TUSACC10.csv) in which we are doing this during runtime/on the fly.



> He sent over this version. However when I went into Dev and tried to process this file in a new Billing Cycle I recieved the "Something went wrong error"

@kdjstudios 
There is a trick to it. If you see there is no header on this  [file (TUSACC10.csv)](http://gitlab.aavaz.biz/AnswerNet/SABilling/uploads/716c51ebf2adf0f55704b438dd3f103d/TUSACC10.csv) so we need to add a header on to this file which will help system identifying which column is which.

Please try this one [TUSACC10.csv](/uploads/4a4077d240c6bc78c451e05e71c84fcf/TUSACC10.csv)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 10, 2017, 08:31

@smasih 

Let me confirm which version of the file they will have access to after they lose their billing system. I have a feeling that the new one is like the HR version of their import from their billing system.

Oh thanks, I did not even notice the missing headers. I will keep that in mind.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 10, 2017, 15:27

Here is the response from Garrett:

The report which you have programmed the import to is the best. It's straight from Szeto. 
So the report that I had sent you in error is not to be used as it's from a report based off our current billing system, thus, when we discontinue our billing system we will stop generating this report. 
Lots of words just to say the original report is the report to use. lol.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Nov 11, 2017, 02:13

@kdjstudios 

Thanks for the update.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 13, 2017, 07:44

@kdjstudios @tim.traylor 

I ran the september billing as well and it seems pretty close, but we have the late charge added on to accounts **(sab accounts)** for about $4700 

Reason: August Invoices are unpaid and net terms is '0' as well.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 13, 2017, 10:25

@smasih @tim.traylor @kace.coady 

So I just ran the Billing cycle detail report for the 9/20 cycle; we have a total of: **364729.01**
Total outlined in Invoice details from SCC: **364372.80**

This is a difference of less then $500. However as stated that is due to the late charges being included; which total **4709.34**; this is still under 2% difference. 

In summary, I believe with confirmation that a majority of the difference is due to the one time charges we should be okay to move forward with a side by side for october?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 14, 2017, 24:24

@kdjstudios 

absolutely, and there must be late charges on their end as well so the difference is even less than **$4709**.   

**and yes, please let's continue with the october billing.**

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 14, 2017, 08:10

@smasih 

May we please confirm that is the cause of those differences before proceeding. We don't want to be making unnecessary assumptions. I will try to put together a list again of all of the accounts that have different balances for you to check the line items to insure they are just the onetime charges and that nothing else major (other then the late fees) are effecting the discrepancy.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 15, 2017, 08:04

@kdjstudios 

I was occupied with moneris integration, so I'll update this ticket tomorrow first thing in the morning.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 22, 2017, 08:12

@smasih @tim.traylor @kace.coady 

According to your post in the October ticket about Septembers billing files the difference file that was uploaded is: [adiff_sep.csv](/uploads/5d878e8ba86c24a04a0130d803f8938d/adiff_sep.csv)

I have added a total to the bottom of the file and there is a difference of roughly 230000. **This is a huge difference.** That is over 60% in discrepancy from the total given by SCC of 364372.

Please find what is causing such a discrepancy, as your original estimate is way off compared to this.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 22, 2017, 12:34

@smasih @tim.traylor @kace.coady 

After a quick totaling of the invoice line items  in the invoice details file, it appears that the file sent over is missing lots of line items. I asked for the originals in pdf format to insure that we are sent all line items.  I will post the updated versions once received here.

This I will hope is the reason there is a 60% discrepancy.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 23, 2017, 08:08

@kdjstudios @tim.traylor @abakshi @kace.coady @pchaudhary 

I checked a couple of accounts that were in difference file and turns out there was no difference in invoice line items for those accounts.

The reason was that **Invoicing_details_.xlsx** wasn't clean at all. I tried cleaning it in order to re-run the difference script. But still due to unorganized **Invoicing_details_.xlsx** I am not able to run script on it.

So I did check approx 30  accounts manually and found no difference (except late charge).

**I don't how to proceed from here onwards, any advice would be appreciated**

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 24, 2017, 08:03

@smasih 

Maybe you didn't understand my post right before yours. But I already stated that the Invoicing details file was missing line items for accounts and that I will be posting an updated version here once I receive it from Garrett.  Once we receive that file, you will then be able to check the accounts and result in an accurate discrepancy.

NOTE: I have not heard from Garrett since Wed, which would lead me believe that he is on holiday for Thanksgiving.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 27, 2017, 11:04

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 27, 2017, 11:05

@smasih @tim.traylor 

Updated invoice details file in the gitlab ticket. Have both the original pdf and one I converted to excell using https://www.pdftoexcel.com/

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 30, 2017, 05:26

@kdjstudios 

here is the difference file for september billing [adiff1.csv](/uploads/cb839e105ac7a8736f61c3964de135b4/adiff1.csv)

The difference total is a lot and in order to resolve this, we need to re-import the data as I mentioned before their data is not constant for each month they keep on adding or removing handles/resource_ids. I will post another difference file on the november ticket after re-importing the data.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 30, 2017, 08:38

@smasih @tim.traylor 

Thank you for the update. And yes that would make sense that since we moved forward with October and updated the accounts that going backwards would cause more discrepancy. So lets discuss the October discrepancies more in depth before we proceed to November.

Marking this ticket for close.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 11, 2018, 13:07

closed
